### ISSUE TITLE
Unable to find a supported build workflow for runtime 'java8'

### ISSUE THREAD

**Issue by seasuresh** [awslabs/aws-sam-cli#1121]
Hi,

It looks like the SAM plugin stopped working on IDEA after upgrading gradle.
I have the lastest SAM plugin already. Any suggestion>

/usr/local/bin/sam build --template /private/var/folders/jw/xxyyufc/T/template5.yaml --build-dir /private/var/folders/jw/xxyyufc/T/lambdaBuild332
2019-04-13 19:54:50 Building resource 'Function'
Build Failed
Error: Unable to find a supported build workflow for runtime 'java8'. Reason: None of the supported manifests '['build.gradle', 'pom.xml']' were found in the following paths '['/Users/mayo/wp/util/src/main', '/private/var/folders/jw/erws/T']'

**Comment by TheSriram** [comment]
Is the `build.gradle` file present in those directories?

**Comment by pietvandongen** [comment]
I have a similar / the same problem. My logs:

```
2019-04-19 09:40:48,176 [ 453940]   INFO - .services.lambda.LambdaBuilder - /usr/local/bin/sam build --template /private/var/folders/1m/5fljzhd51d3d9z80lcyyb6c40000gn/T/template10.yaml --build-dir /private/var/folders/1m/5fljzhd51d3d9z80lcyyb6c40000gn/T/lambdaBuild11

2019-04-19 09:40:48,608 [ 454372]   INFO - rationStore.ComponentStoreImpl - Saving Project '/Users/pietvandongen/Repository/room-unit-server' room-unit-serverMacroExpansionManager took 14 ms
2019-04-19 09:40:48,621 [ 454385]   INFO - mponents.impl.stores.StoreUtil - saveProjectsAndApp took 92 ms
2019-04-19 09:40:48,680 [ 454444]   INFO - .services.lambda.LambdaBuilder -

2019-04-19 09:40:48,680 [ 454444]   INFO - .services.lambda.LambdaBuilder - 2019-04-19 09:40:48 Building resource 'Function'

2019-04-19 09:40:48,680 [ 454444]   INFO - .services.lambda.LambdaBuilder - Error: Unable to find a supported build workflow for runtime 'java8'. Reason: None of the supported manifests '['build.gradle', 'build.gradle.kts', 'pom.xml']' were found in the following paths '['/Users/pietvandongen/Repository/room-unit-alexa-smart-home-skill/src/main', '/private/var/folders/1m/5fljzhd51d3d9z80lcyyb6c40000gn/T']'

2019-04-19 09:40:48,680 [ 454444]   INFO - .services.lambda.LambdaBuilder - Build Failed

2019-04-19 09:40:48,714 [ 454478]   WARN - xecution.local.SamInvokeRunner - Failed to create Lambda package
java.util.concurrent.CompletionException: java.lang.IllegalStateException: SAM build command failed
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:647)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at software.aws.toolkits.jetbrains.services.lambda.LambdaBuilderUtils$buildAndReport$$inlined$runSamBuildInBackground$1$1.apply(LambdaBuilderUtils.kt:108)
	at software.aws.toolkits.jetbrains.services.lambda.LambdaBuilderUtils$buildAndReport$$inlined$runSamBuildInBackground$1$1.apply(LambdaBuilderUtils.kt:97)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:797)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at software.aws.toolkits.jetbrains.services.lambda.LambdaBuilder$buildLambdaFromTemplate$2$2.processTerminated(LambdaBuilder.kt:126)
	at sun.reflect.GeneratedMethodAccessor179.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:214)
	at com.sun.proxy.$Proxy22.processTerminated(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.lambda$notifyTerminated$2(ProcessHandler.java:172)
	at com.intellij.execution.process.ProcessHandler$TasksRunner.execute(ProcessHandler.java:247)
	at com.intellij.execution.process.ProcessHandler.notifyTerminated(ProcessHandler.java:155)
	at com.intellij.execution.process.ProcessHandler.notifyProcessTerminated(ProcessHandler.java:151)
	at com.intellij.execution.process.KillableProcessHandler.notifyProcessTerminated(KillableProcessHandler.java:123)
	at com.intellij.execution.process.BaseProcessHandler.onOSProcessTerminated(BaseProcessHandler.java:61)
	at com.intellij.execution.process.OSProcessHandler.onOSProcessTerminated(OSProcessHandler.java:112)
	at com.intellij.execution.process.BaseOSProcessHandler$1.lambda$startNotified$0(BaseOSProcessHandler.java:99)
	at com.intellij.execution.process.ProcessWaitFor.lambda$null$0(ProcessWaitFor.java:61)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:224)
	at com.intellij.execution.process.ProcessWaitFor.lambda$new$1(ProcessWaitFor.java:43)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: SAM build command failed
	... 22 more
```

My output of `sam --version`:

```
SAM CLI, version 0.15.0
```

The IntelliJ plugin version of AWS Toolkit is `v.1.2`.

**Comment by pietvandongen** [comment]
By looking at the logs it seems that SAM CLI is looking for my `build.gradle` file in the wrong directory:

```
Reason: None of the supported manifests '['build.gradle', 'build.gradle.kts', 'pom.xml']' were found in the following paths '['/Users/pietvandongen/Repository/room-unit-alexa-smart-home-skill/src/main', '/private/var/folders/1m/5fljzhd51d3d9z80lcyyb6c40000gn/T']'
```

My `build.gradle` file is in the project root, so at:

```
/Users/pietvandongen/Repository/room-unit-alexa-smart-home-skill/build.gradle
```

**Comment by pietvandongen** [comment]
So this seems to be linked to this issue:

https://github.com/aws/aws-toolkit-jetbrains/issues/857

I now have a successful workaround where I have a `template.yaml` in the root of my project directory (at the same level as the `build.gradle` file). I then point to template in the AWS Toolkit run configuration:

My `template.yaml` looks like this:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: .
      Handler: com.bdrthermea.remote.application.externalapi.smarthomeskill.SmartHomeDirectiveHandler::handleRequest
      Runtime: java8
```

**Comment by jfuss** [comment]
This was an issue within the AWS toolkit and was resolved: aws/aws-toolkit-jetbrains#857

Closing.
